controller/workitem_blackbox_test.go

```diff
@@ -1659,6 +1660,331 @@ func (s *WorkItem2Suite) TestWI2ListByStateFilterOKModifiedUsingIfNoneMatchIfMod
+				fxt.WorkItems[5].ID,
+				fxt.WorkItems[6].ID,
+			}.ToMap()
+			require.Equal(t, fxt.WorkItems[0].ID, *actualWIs.Data[6].ID)
+			require.Equal(t, fxt.WorkItems[1].ID, *actualWIs.Data[5].ID)
+			require.Equal(t, fxt.WorkItems[2].ID, *actualWIs.Data[4].ID)
+			require.Equal(t, fxt.WorkItems[3].ID, *actualWIs.Data[3].ID)
+			require.Equal(t, fxt.WorkItems[4].ID, *actualWIs.Data[2].ID)
+			require.Equal(t, fxt.WorkItems[5].ID, *actualWIs.Data[1].ID)
+			require.Equal(t, fxt.WorkItems[6].ID, *actualWIs.Data[0].ID)
+			for _, wi := range actualWIs.Data {
+				_, ok := toBeFound[*wi.ID]
+				require.True(t, ok, "unknown work item found: %s", wi.ID)
+				delete(toBeFound, *wi.ID)
+			}
+			require.Empty(t, toBeFound, "failed to found all work items: %+s", toBeFound)
+		})
+		t.Run("by created ascending", func(t *testing.T) {
+			exp := ptr.String(`{"system.state": "open"}`)
+			sort := ptr.String("created")
+			_, actualWIs := test.ListWorkitemsOK(s.T(), s.svc.Context, s.svc, s.workitemsCtrl, fxt.Spaces[0].ID, exp, nil, nil, nil, nil, nil, nil, nil, nil, nil, sort, nil, nil)
+			// then
+			require.NotNil(s.T(), actualWIs)
+			require.Len(s.T(), actualWIs.Data, 7)
+			toBeFound := id.Slice{
+				fxt.WorkItems[0].ID,
+				fxt.WorkItems[1].ID,
+				fxt.WorkItems[2].ID,
+				fxt.WorkItems[3].ID,
+				fxt.WorkItems[4].ID,
+				fxt.WorkItems[5].ID,
+				fxt.WorkItems[6].ID,
+			}.ToMap()
+			require.Equal(t, fxt.WorkItems[0].ID, *actualWIs.Data[0].ID)
+			require.Equal(t, fxt.WorkItems[1].ID, *actualWIs.Data[1].ID)
+			require.Equal(t, fxt.WorkItems[2].ID, *actualWIs.Data[2].ID)
+			require.Equal(t, fxt.WorkItems[3].ID, *actualWIs.Data[3].ID)
+			require.Equal(t, fxt.WorkItems[4].ID, *actualWIs.Data[4].ID)
+			require.Equal(t, fxt.WorkItems[5].ID, *actualWIs.Data[5].ID)
+			require.Equal(t, fxt.WorkItems[6].ID, *actualWIs.Data[6].ID)
+			for _, wi := range actualWIs.Data {
+				_, ok := toBeFound[*wi.ID]
+				require.True(t, ok, "unknown work item found: %s", wi.ID)
+				delete(toBeFound, *wi.ID)
+			}
+			require.Empty(t, toBeFound, "failed to found all work items: %+s", toBeFound)
+		})
+
+		t.Run("by updated descending", func(t *testing.T) {
+			// when
+			payload := app.UpdateWorkitemPayload{
+				Data: &app.WorkItem{
+					Type: APIStringTypeWorkItem,
+					ID:   &fxt.WorkItems[3].ID,
+					Attributes: map[string]interface{}{
+						workitem.SystemState:   "resolved",
+						workitem.SystemVersion: fxt.WorkItems[3].Version,
+					},
+				},
+			}
+
+			test.UpdateWorkitemOK(s.T(), s.svc.Context, s.svc, s.workitemCtrl, fxt.WorkItems[3].ID, &payload)
+
+			payload = app.UpdateWorkitemPayload{
+				Data: &app.WorkItem{
+					Type: APIStringTypeWorkItem,
+					ID:   &fxt.WorkItems[2].ID,
+					Attributes: map[string]interface{}{
+						workitem.SystemState:   "resolved",
+						workitem.SystemVersion: fxt.WorkItems[2].Version,
+					},
+				},
+			}
+
+			test.UpdateWorkitemOK(s.T(), s.svc.Context, s.svc, s.workitemCtrl, fxt.WorkItems[2].ID, &payload)
```

Comment:
Please move all the `updateWorkItemOk` into a `for` loop

Reply:
Fixed. 4ed2790ae282930224628db5ac0cb008bb3d489f